### PR TITLE
Fixes #9163: Add an option to add timing information to the agent run

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9163

### CODE REVIEW

**Comment by amousset** [comment]
Currently not mergeable as the *-t* option clashes with token, do you have an idea of another name for the option, or another letter?

**Comment by amousset** [comment]
Commit modified

**Comment by peckpeck** [comment]
In which command ?
An answer could be -T

**Comment by amousset** [comment]
Every command that start an agent, namely: run, inventory, remote-run and factory-reset. I'll go with -T.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: share/lib/reports.awk [hunk 1]
```diff
@@ -42,6 +43,15 @@ function alen (a) {
   return k;
 }
 
+function timer() {
+  "date +%s.%N" | getline component_end_time;
+  close("date +%s.%N");
+  tmp_time = component_end_time - component_begin_time;
+  "date +%s.%N" | getline component_begin_time;
```
Comment (peckpeck) [inline share/lib/reports.awk]: why call date twice, with only a substraction between them ?

File: share/lib/reports.awk [hunk 2]
```diff
@@ -32,6 +32,7 @@ BEGIN {
   "date +%s.%N" | getline starttime;
   # needed to be able to call the same command a second time
   close("date +%s.%N");
+  timer()
```
Comment (peckpeck) [inline share/lib/reports.awk]: Comment to say why you have to call it here

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Pull request merged**